Add API version 2025-05-01 for dnsresolver (#49921)
Azure/azure-sdk-for-net: sdk/dnsresolver/Azure.ResourceManager.DnsResolver/CHANGELOG.md

# Release History

> ## 2.0.0 (2025-06-06)

Reviewer: we cannot have this change.
Mgmt libraries have a principal that all the mgmt libraries should share the same major version therefore unless very rare case happens, we would never bump major versions for one single library.
Therefore please revert this change, and we will provide guidance for you to use some C# tricks to mitigate the breaking change on the API level.

Author: Got it, thank you I will revert back to 1.x

Author: Rverted

### Features Added
- Upgraded api-version tag from 'package-preview-2023-07' to 'package-2025-05'. Tag detail available at https://github.com/Azure/azure-rest-api-specs/blob/8600539fa5ba6c774b4454a401d9cd3cf01a36a7/specification/dnsresolver/resource-manager/readme.md.
- Adds new POST bulk API for large domain list usage.

### Breaking Changes

> - BlockResponseCode has been removed from the DnsSecurityRule Action type and can no longer be specified.

Reviewer: ```suggestion
- BlockResponseCode has been removed from the DnsSecurityRule Action type in api-version `2025-05-01`.
```

Author: Changed
